COMMITTERS.adoc

```diff
@@ -0,0 +1,182 @@
+  better ping them again to have some information
+  or ask the original author if we should go ahead.
+* If in doubt (on a specific part or because you're
+  not comfortable reviewing documentation or any other reason),
+  ask for help
+* Except if it's a total no-brainer (typically a typo fix),
+  let the PR bake a few hours so some other people can have a
+  look if interested.
+  If the PR is very large or is a new extension, better let it
+  bake for a few days.
+* We do merge commits so please ask the author to properly
+  squash the commits before merging.
+  The idea is to have proper semantic commits.
+  If the author is not familiar with Git or not available,
+  you can also (carefully)
+  https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork[do it for them]
+  as long as you agreed on it.
+* Make sure the commit comments are meaningful.
+* As a new committer, you can start by giving your approval and not merging
+  and see how things evolve.
+  This should give you some hints of important things to check in the future.
+* Only merge pull requests that target master.
+  **Dealing with old branches is a separate process.**
+* If you are sure a PR should go in and is just waiting for CI,
+  just use the `triage/waiting-for-ci` label.
+  It is a good hint for other committers.
+* If you think a pull request is an essential part of a given milestone
+  (and can realistically be part of it),
+  you should affect this milestone to it so that we can be sure it stays on the radar.
+
+Obviously, each situation is different so use your own judgement,
+and if in doubt, just ask for advice, the other committers are here to help.
+
+== Pull Request Title
+
+The title should be in English, should not contain an issue number.
+It should also not contain ellipsis.
+
+If your commit message was too long and GitHub automatically cut
+the title, that's helpful if you can take the time to move the cut
+part where it belongs to have a full title.
+
+Titles are included in the Release notes so they are important.
+
+A good title would look like: `Fix XXX in Quartz extension` or
+`Introduce Hibernate Reactive extension`.
+
+A bad title would look like: `fix(#444)`.
+
+== Issues Fixed
+
+When a PR fixes some issues, it's good practice to add it in the description.
+
+One issue per line with something like:
+
+[source,asciidoc]
+----
+* Fix #444
+* Fix #555
+----
+
+[TIP]
+====
+GitHub supports a variety of keywords here: `fix`, `fixes`, `fixed`,
+`resolve`, `resolves`, `resolved`, `close`, `closes`, `closed`
+all do the same thing.
+====
+
+[WARNING]
+====
+GitHub won't detect issues properly if you do something like
+`Fix #444 #555`.
+====
+
+== Affecting Labels and Milestones
+
+Affecting labels and milestones is very important in our process.
+
+Before each release, we check that we don't have any orphan i.e.
+closed issues that don't have either a milestone affected or
```

Comment:
this part we can definitely automate us out of - just have a check on close that labels and milestones are applied properly.